src/CddInterface.c

```diff
@@ -278,23 +257,20 @@ static long int *ddG_InteriorPoint(dd_MatrixPtr M)
   set_copy(RR, R);         /* copy of R. */
   set_copy(SS, S);         /* copy of S. */
 
+  Obj result;
+
   if (dd_ExistsRestrictedFace(M, R, S, &err))
   {
     set_uni(M->linset, M->linset, R);
     dd_FindRelativeInterior(M, &ImL, &Lbasis, &lps, &err);
     dim = M->colsize - set_card(Lbasis) - 1;
     set_uni(M->linset, M->linset, ImL);
-    result[0] = dim;
+
+    result = NEW_PLIST(T_PLIST_EMPTY, lps->d);
+    ASS_LIST(result, 1, INTOBJ_INT(dim));
     for (j = 1; j < (lps->d) - 1; j++)
     {
-
-      mpq_get_num(u, lps->sol[j]);
-      mpq_get_den(v, lps->sol[j]);
-      z1 = mpz_get_si(u);
-      z2 = mpz_get_si(v);
-
-      result[2 * (j - 1) + 1] = z1;
-      result[2 * (j - 1) + 2] = z2;
+      ASS_LIST(result, j + 1, MPQ_TO_GAPOBJ(lps->sol[j]));
```

Comment:
Not only is this much simpler, and avoids passing data through a global variable (always bad style), it also works correctly with data that does not fit into a GAP intobj.